src/lighteval/tasks/registry.py

```diff
@@ -142,22 +124,89 @@ def get_task_dict(
             hlog(custom_tasks_registry)
 
+        default_tasks_registry = create_config_tasks(cache_dir=self._cache_dir)
+        # Check the overlap between default_tasks_registry and custom_tasks_registry
+        intersection = set(default_tasks_registry.keys()).intersection(set(custom_tasks_registry.keys()))
+        if len(intersection) > 0:
+            hlog_warn(
+                f"Following tasks ({intersection}) exists both in the default and custom tasks. Will use the default ones on conflict."
+            )
+
+        # Defaults tasks should overwrite custom tasks
+        return {**default_tasks_registry, **custom_tasks_registry}
+
+    @property
+    @lru_cache
+    def _task_superset_dict(self):
+        """
+        Returns a dict mapping each superset to its tasks:
+        eg: mmlu -> [mmlu:abstract_algebra, mmlu:college_biology, ...]
+        """
+
+        superset_dict = {k: list(v) for k, v in groupby(list(self.task_registry.keys()), lambda x: x.split(":")[0])}
+        # Only consider supersets with more than one task
+        return {k: v for k, v in superset_dict.items() if len(v) > 1}
+
+    @property
+    @lru_cache
+    def task_groups_dict(self) -> dict[str, str]:
+        """
+        Returns a dict mapping each task group to its tasks
+        eg: all_cusotm -> custom|task1,custom|task2,custom|task3
+        """
+        if self._custom_tasks is None:
+            return {}
+        custom_tasks_module = create_custom_tasks_module(custom_tasks=self._custom_tasks)
+        tasks_group_dict = {}
+        if hasattr(custom_tasks_module, "TASKS_GROUPS"):
+            tasks_group_dict = custom_tasks_module.TASKS_GROUPS
+        return tasks_group_dict
+
+    def get_task_dict(self, task_names: list[str]) -> dict[str, LightevalTask]:
+        """
+        Get a dictionary of tasks based on the task name list.
+
+        Args:
+            task_name_list (List[str]): A list of task names.
+            custom_tasks (Optional[Union[str, ModuleType]]): Path to the custom tasks file or name of a module to import containing custom tasks or the module itself
+            extended_tasks (Optional[str]): The path to the extended tasks group of submodules
+
+        Returns:
+            Dict[str, LightevalTask]: A dictionary containing the tasks.
+
+        Notes:
+            - If custom_tasks is provided, it will import the custom tasks module and create a custom tasks registry.
+            - Each task in the task_name_list will be instantiated with the corresponding task class.
+        """
         # Select relevant tasks given the subset asked for by the user
-        tasks_dict = {}
-        for task_name in task_name_list:
-            task_class = self.get_task_class(task_name, custom_tasks_registry=custom_tasks_registry)
-            tasks_dict[task_name] = task_class()
+        return {task_name: self.get_task_class(task_name)() for task_name in task_names}
```

Comment:
Not super fond of the "return a function then call it" of `self.get_task_class(task_name)()` because it reduces readability. If there's an easy way to simplify would be great.